### ISSUE TITLE
Supplemental Claims | Sidekiq 4142 form

### ISSUE THREAD

**Issue by saderagsdale** [department-of-veterans-affairs/va.gov-team#61012]
## Value Statement
**_As a_** Veteran submitting a supplemental claims with 4142 information in the payload
**_I want to_** be sure that my 4142 form gets generated and sent to the VA
**_So that_** the VA can process it and get my medical records. 

---
## Background Context
- Development effort from #58838 

## Acceptance Criteria
- [ ] ~Initial attempt is sycronous as it currently is~
- [x] ~Upon failure,~ a sidekiq job is queued to retry some number of time (lets try 3)
- [x] Logging is added to see when this happens

## Designs and Build Notes
- Add here

## Out of scope
- Resolving platform issue causing original failures

## Open questions
- Add here

## Tasks
- [x] Add code
- [x] Add test case
- [x] Open PR
- [ ] Merge
- [ ] Monitor 

## Definition of Ready 

- [x] Clear value description
- [x] Testable acceptance criteria
- [x] Accessibility added to acceptance criteria
- [x] Approved designs attached
- [x] Sample data provided where appropriate
- [x] Estimated to fit within the sprint
- [x] Dependencies and blockers linked

## Definition of Done

- [x] Meets acceptance criteria
- [x] Passed unit testing (90% coverage)
- [x] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Monitor production for occurrences and proper functionality

**Comment by saderagsdale** [comment]
@kylesoskin we can add details about the sidekiq job here.

**Comment by kylesoskin** [comment]
@saderagsdale I have instead made this ALWAYS asynchronous. Meaning every 4142 will be a queued job. 

This is for a few reasons but mainly: 

- This is how it is done elsewhere (526)
- This gets the response/return back to the user quicker in all cases, directly impacting the goal of a more performant website and better veteran experience. 

IE it used to be:

```
send_submission
generate_and_send_4142 # blocking, had to wait for pdf to generate, and external http call to CMP to come back
return response_to_frontend_user
```
now it is going to be
```
send_submission
queue_generate_and_send_4142 # non-blocking, comes back right away, plops job in queue
return response_to_frontend_user
```


Let me know if you have any thoughts on this.

**Comment by saderagsdale** [comment]
@kylesoskin Is there a notable difference in performance between async and synch? Also, any idea why async is the go-to pattern (the answer to the first question might answer this one)?

**Comment by kylesoskin** [comment]
> Is there a notable difference in performance between async and synch

For the end user, yes, between 1-5s (estimated, sometimes much more, or slightly less) on average to get to the confirmation page, in the current synch way. Because it has to wait for the pdf to generate, get sent, and get a response back from Central mail. The website for the user is just spinning/loading at that time while it is waiting for that, before proceeding to the confirmation page. 

By switching to async, we eliminate that waiting time. Because queuing it to be sent take almost no time at all (to put in our queue). 


Async is the go to (in my opinion) for either unreliable third party services, or for actions the user does not explicitly know are happening (IE why make them wait on something they may not know is happening or already thought happened).

**Comment by kylesoskin** [comment]
This is merged, and confirmed working in productions, based on the logs we can see the jobs being queued and succeeding.
